### PR TITLE
AWS: Switch tests to Junit5

### PR DESCRIPTION
**Pull request by skytin1004**
This pull request is related to the issue "Move JUnit4 tests to JUnit5 https://github.com/apache/iceberg/issues/7160".
I've switched the files in AWS pakage to JUnit5.

### CODE REVIEW

File: aws/src/test/java/org/apache/iceberg/aws/glue/TestGlueCatalog.java [hunk 1]
```diff
@@ -602,17 +601,18 @@ public void testTablePropsDefinedAtCatalogLevel() {
...
-    Assert.assertEquals("catalog-default-key2", properties.get("table-default.key2"));
-    Assert.assertTrue(properties.containsKey("table-default.key3"));
-    Assert.assertEquals("catalog-default-key3", properties.get("table-default.key3"));
-    Assert.assertTrue(properties.containsKey("table-override.key3"));
-    Assert.assertEquals("catalog-override-key3", properties.get("table-override.key3"));
-    Assert.assertTrue(properties.containsKey("table-override.key4"));
-    Assert.assertEquals("catalog-override-key4", properties.get("table-override.key4"));
+    Assertions.assertThat(properties)
+        .isNotEmpty()
+        .containsKey("table-default.key1")
```
Comment (nastra) [inline aws/src/test/java/org/apache/iceberg/aws/glue/TestGlueCatalog.java:606]: we can omit all the `containsKey` calls for which we also check `containsEntry`

File: aws/src/test/java/org/apache/iceberg/aws/glue/TestGlueCatalog.java [hunk 2]
```diff
@@ -652,19 +653,17 @@ public void testTableLevelS3TagProperties() {
...
+        .containsKey(
+            AwsProperties.S3_WRITE_TAGS_PREFIX.concat(AwsProperties.S3_TAG_ICEBERG_NAMESPACE));
+    Assertions.assertThat(
+            tableCatalogProperties.get(
+                AwsProperties.S3_WRITE_TAGS_PREFIX.concat(AwsProperties.S3_TAG_ICEBERG_TABLE)))
+        .isEqualTo("table");
+    Assertions.assertThat(
+            tableCatalogProperties.get(
+                AwsProperties.S3_WRITE_TAGS_PREFIX.concat(AwsProperties.S3_TAG_ICEBERG_NAMESPACE)))
+        .isEqualTo("db");
```
Comment (nastra) [inline aws/src/test/java/org/apache/iceberg/aws/glue/TestGlueCatalog.java:667]: better to use containsEntry here and above

**Review by nastra — APPROVED** [review]

**Comment by skytin1004** [comment]
Thanks @nastra, I have fixed the code based on your review.

File: aws/src/test/java/org/apache/iceberg/aws/glue/TestGlueCatalog.java (hunk 1 above)
Comment (skytin1004) [reply aws/src/test/java/org/apache/iceberg/aws/glue/TestGlueCatalog.java]: I've omit all the containsKey

**Pull request merged**